src/components/CallView/PromotedView/PromotedView.vue

```diff
@@ -0,0 +1,477 @@
+}
+
+.videoContainer.hidden,
+.participants-1.screensharing .videoContainer.hidden,
+.participants-2.screensharing .videoContainer.hidden {
+	display: none;
+}
+
+.screensharing .videoContainer {
+	max-height: 200px;
+}
+
+.constrained-layout.screensharing .videoContainer {
+	max-height: 100px;
+
+	/* Avatars slightly overflow the container; although they overlap the shared
+	 * screen it is not too bad and it is better than compressing even further
+	 * the shared screen. */
+	overflow: visible;
+}
+
+::v-deep video {
+	z-index: 0;
+	max-height: 100%;
+	/* default filter for slightly better look */
+	/* Disabled for now as it causes a huuuuge performance drop.
+	 CPU usage is more than halved without this.
+	 -webkit-filter: contrast(1.1) saturate(1.1) sepia(.1);
+	 filter: contrast(1.1) saturate(1.1) sepia(.1);
+	 */
+	vertical-align: top; /* fix white line below video */
+}
+
+.screensharing .videoContainer ::v-deep video {
+	max-height: 200px;
+	background-color: transparent;
+	box-shadow: none;
+}
+
+#screens ::v-deep video {
+	width: 100%;
+	-webkit-filter: none;
+	filter: none;
+}
+
+#videos .videoContainer.not-connected ::v-deep {
+	video,
+	.avatardiv,
+	.avatar.guest {
+		opacity: 0.5;
+	}
+}
+
+#videos .videoContainer:not(.promoted) ::v-deep video {
+	max-height: 200px;
+	max-width: 100%;
+	background-color: transparent;
+	border-radius: var(--border-radius) var(--border-radius) 0 0;
+	box-shadow: 0 0 15px var(--color-box-shadow);
+	bottom: 44px;
+	position: relative;
+}
+
+.constrained-layout #videos .videoContainer:not(.promoted) ::v-deep video {
+	/* Make the unpromoted videos smaller to not overlap too much the promoted
+	 * video */
+	max-height: 100px;
+}
+
+#videos .videoContainer ::v-deep .avatardiv {
+	box-shadow: 0 0 15px var(--color-box-shadow);
+}
+
+.participants-1 #videos .videoContainer ::v-deep video,
+.participants-2 #videos .videoContainer ::v-deep video {
+	padding: 0;
+}
+
+.videoContainer ::v-deep .avatar-container {
+	position: absolute;
+	bottom: 44px;
+	left: 0;
+	width: 100%;
+}
+
+.videoContainer ::v-deep .avatar-container .avatardiv {
+	display: block;
+	margin-left: auto;
+	margin-right: auto;
+}
+.videoContainer.promoted ::v-deep .avatar-container {
+	top: 30%;
+}
+.videoContainer.promoted ::v-deep .avatar-container + .nameIndicator {
+	display: none;
+}
+
+.videoContainer.promoted ::v-deep .mediaIndicator {
+	display: none !important;
+}
+
+.participants-1:not(.screensharing) ~ #emptycontent {
+	display: block !important;
+}
+
+/* big speaker video */
+.participants-1 .videoContainer,
+.participants-2 .videoContainer,
+.videoContainer.promoted {
+	position: absolute;
+	width: 100%;
+	height: 100%;
+	overflow: hidden;
+	left: 0;
+	top: 0;
+	z-index: 1;
+}
+.videoContainer.promoted ::v-deep video,
+.participants-2 .videoContainer:not(.videoView) ::v-deep video {
+	position: absolute;
+	width: initial;
+	height: 100%;
+	left: 50%;
+	top: 50%;
+	transform: translateY(-50%) translateX(-50%);
```

Comment:
You should center with `object-cover`, not `transform`

Reply:
Form what I read in mdn this is unrelated

Reply:
That is not what I mean, good habit in css is to avoid absolute to the max you can. Additionally, trasnfrom is a browser rendering function that is expensive. Centering with the dedicated object-fit function is what you need to do.

Reply:
Please not though that it doesn't work with ie, which is fine, you will just see black bars. But it's ie and we deprecate for nc20
https://caniuse.com/#feat=object-fit

Reply:
Fit modifies height and width of the element, there I only need to center it into it's container so this doesn't seem to be an alternative.. Height and width of the video are computed manually depending on the video aspect ratio and the parent aspect ratio.

Reply:
This is how the computation is done: https://github.com/nextcloud/spreed/blob/df4570615b87792b240557e60e1cb0e6013d306d/src/components/CallView/shared/video.js#L64